integration/webhook_test.go

```diff
@@ -0,0 +1,175 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	k8sApiErrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"github.com/openshift/addon-operator/integration"
+)
+
+const (
+	CATALOG_SOURCE_URL = "quay.io/osd-addons/reference-addon-index@sha256:58cb1c4478a150dc44e6c179d709726516d84db46e4e130a5227d8b76456b5bd"
+	ADDON_NAME         = "reference-addon-test-install-spec"
+)
+
+func TestAddonInstallSpec(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	testCases := []struct {
+		addon *addonsv1alpha1.Addon
+		err   *k8sApiErrors.StatusError
+	}{
+		{
+			addon: newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+				Type: addonsv1alpha1.OLMOwnNamespace,
+			}),
+			err: newStatusError(".spec.install.ownNamespace is required " +
+				"when .spec.install.type = OwnNamespace"),
+		},
+		{
+			addon: newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+				Type: addonsv1alpha1.OLMAllNamespaces,
+			}),
+			err: newStatusError(".spec.install.allNamespaces is required " +
+				"when .spec.install.type = AllNamespaces"),
+		},
+		{
+			addon: newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+				Type: addonsv1alpha1.OLMOwnNamespace,
+				OLMOwnNamespace: &addonsv1alpha1.AddonInstallOLMOwnNamespace{
+					AddonInstallOLMCommon: addonsv1alpha1.AddonInstallOLMCommon{
+						Namespace:          "reference-addon",
+						PackageName:        ADDON_NAME,
+						Channel:            "alpha",
+						CatalogSourceImage: CATALOG_SOURCE_URL,
+					},
+				},
+			}),
+			err: nil,
+		},
+		{
+			addon: newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+				Type: addonsv1alpha1.OLMAllNamespaces,
+				OLMAllNamespaces: &addonsv1alpha1.AddonInstallOLMAllNamespaces{
+					AddonInstallOLMCommon: addonsv1alpha1.AddonInstallOLMCommon{
+						Namespace:          "reference-addon",
+						PackageName:        ADDON_NAME,
+						Channel:            "alpha",
+						CatalogSourceImage: CATALOG_SOURCE_URL,
+					},
+				},
+			}),
+			err: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := integration.Client.Create(ctx, tc.addon)
+
+		if tc.err != nil {
+			if !reflect.DeepEqual(err, tc.err) {
+				log.Fatalf("unexpected error: %v\nexpected: %v",
```

Comment:
Please replace these with either t.Fatal, or better try to use the assert/require functions from: https://pkg.go.dev/github.com/stretchr/testify